internal/provider/utils.go

```diff
@@ -1231,7 +1232,32 @@ func uploadFile(url, filePath string, formFields map[string]any) error {
 	client := &http.Client{
 		Timeout: 20 * time.Minute,
 	}
-	_, err = sling.New().Client(client).Base(url).Set("Content-Type", writer.FormDataContentType()).Post("").Body(&buffer).ReceiveSuccess(nil)
+	if cloud == "AZURE" && isFlinkArtifact {
+		var contentFormat string
+		switch fileExtension {
```

Comment:
nit: I must be missing something, but it seems like there's no tolower() being called for fileExtension variable and this attribute is in uppercase in the response ("content_format": "JAR" )

Reply:
but given that e2e example does work, I'm likely missing something 😅